packages/app-desktop/gui/note-viewer/index.html

```diff
@@ -589,6 +589,22 @@
 			let element = event.target;
 
 			// To handle right clicks on resource icons
+			let mermaidElement = element.closest('.mermaid');
+			if(mermaidElement) {
+				const svgString = new XMLSerializer().serializeToString(mermaidElement);
+				// Remove any characters outside the Latin1 range
+				const decoded = unescape(encodeURIComponent(svgString));
+				// Now we can use btoa to convert the svg to base64
+				const base64 = btoa(decoded);
+				const dataUrl = `data:image/svg+xml;base64,${base64}`;
+				ipcProxySendToHost('contextMenu', {
+						type: 'resource',
+						resourceContent: svgString,
+						resourceURL: dataUrl,
```

Comment:
Should be `resourceDataUrl`.

Why do we need both resourceContent and resourceURL?

Reply:
The url needed by copyToClipboard is basically `data:image/svg+xml;base64,{base64 of resourceContent}` but we cannot just append `data:image/svg+xml;base64` in front the data blindly in the contextMenu because then it won't work for other formats that we might integrate in future, thus not a generalised solution. Let me know if you find a better way to do this. We need the url for `Copy path to clipboard` feature.

Reply:
I can't follow the logic sorry, but I think that's because things are not named properly. resourceUrl is the file content apparently. But there's also resourceContent, which is... what? Also the resource content? So why are both needed.

Reply:
And you do this all even before the user asked to export for every right click on a diagram.

Reply:
>  resourceUrl is the file content apparently. But there's also resourceContent, which is... what? Also the resource content? So why are both needed.

Both contains the file data yes, but in different formats, one is suited for saving to a local file (resourceContent), where as another (resourceUrl) is for web compatible [data URI](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs). We can generate the data URI from `resourceContent` but we need the mime type for this (we can send mime type instead and generate it, its a decision choice we need to make)

Reply:
> And you do this all even before the user asked to export for every right click on a diagram.

The svg is generated from the rendered html, the webview and contextMenu handler communicates over ipc, one way to fix this could be to send only the element id and somehow extract the xml version of the svg either over another ipc call or from somewhere where the note html is accessible, this will sure cause some extra code and infra. Let me know if its something vital, then I will look into it in more detail how to do it.

Reply:
I think Roman is correct - you're processing and converting data here even though that might not be needed. You should transfer only the absolute minimum, which would be the svg data (and possibly the mime type is you need that).

Then if you need it as dataURL, you should convert only when it's needed. Also please create a function that does this (for example we already have imageToDataUrl).

Reply:
Cool that sorts it out, thanks! will make the changes soon.